src/config.ts

```diff
@@ -149,7 +149,7 @@ export function createProxyConfig(option: IProxyOption): IProxyConfig {
         customStrategies,
         clientKeys,
         proxyBasePath:
-            option.proxyBasePath || process.env.PROXY_BASE_PATH || '',
+            option.proxyBasePath || process.env.PROXY_BASE_PATH || '/proxy',
```

Comment:
I general I agree with the suggestion on making the "/proxy" suffix optional. But doing it like ends up as a breaking change, as it will no longer assumes the "/proxy" as a suffix to the proxyBasePath. 

We can instead can introduce a `proxyBasePathSuffix` which is defaulting to '/proxy' but can be set to whatever you want, e.g. an empty string `''`;

Reply:
Well, actually your documentation is misleading and describes exactly the behaviour which I'm proposing:

`proxyBasePath : The base path to run the proxy from. Defaults to "/proxy"`

It doesn't mention any "suffix" but the entire `proxyBasePath` parameter. I've lost a couple of working hours fighting with this confusing parameter until I've checked the code. So I'm making the tool work exactly as defined by the documentation.

If you want to keep the current (incorrect) behavior then you should update the documentation accordingly.

Reply:
Fair point, the documentation is also wrong. 

> I've lost a couple of working hours fighting with this confusing parameter until I've checked the code.

I am sorry for that. We try our best to make everything correct. Sometimes we do mistakes, as you probably also have done in your past. 

> If you want to keep the current (incorrect) behavior then you should update the documentation accordingly.

Of course we do not want the documentation to be wrong, and the implementation should be "correct" and as "simple as possible to understand". But we also have to consider that people are already using the proxy in production, and it can cause them quite lot of problems if we just fix this out of the blue. 

After reading your feedback, and reflecting a bit more, I think we do not need to the suffix part. What we can do is simply introduce a new option that works "correctly" and deprecate the use of "PROXY_BASE_PATH", but keep it as it is for a few version. This would allow us to warn users about the change (simple logger.warn message), and still provide a correct implementation. 

I suggest we call the new option `PROXY_PATH`. We will still support `PROXY_BASE_PATH`, it will work as today, and will sett the "proxyPath" option. If you decide to use `PROXY_PATH` it should work according to the documentation, and take precedence. 

What do you think?

Reply:
I like it a lot more. Sounds like a robust solution.

Reply:
Would you like to take a swing on it @danil-smirnov, or should we take over the PR?

Reply:
@ivarconr I'm not sure I have time to go deep with this PR at this moment.